Carbon.Permissions/Functions/Test-CPermission.ps1

```diff
@@ -0,0 +1,232 @@
+
+    .LINK
+    Grant-CPermission
+
+    .LINK
+    Revoke-CPermission
+
+    .LINK
+    http://msdn.microsoft.com/en-us/library/system.security.accesscontrol.filesystemrights.aspx
+
+    .LINK
+    http://msdn.microsoft.com/en-us/library/system.security.accesscontrol.registryrights.aspx
+
+    .LINK
+    http://msdn.microsoft.com/en-us/library/system.security.accesscontrol.cryptokeyrights.aspx
+
+    .EXAMPLE
+    Test-CPermission -Identity 'STARFLEET\JLPicard' -Permission 'FullControl' -Path 'C:\Enterprise\Bridge'
+
+    Demonstrates how to check that Jean-Luc Picard has `FullControl` permission on the `C:\Enterprise\Bridge`.
+
+    .EXAMPLE
+    Test-CPermission -Identity 'STARFLEET\GLaForge' -Permission 'WriteKey' -Path 'HKLM:\Software\Enterprise\Engineering'
+
+    Demonstrates how to check that Geordi LaForge can write registry keys at `HKLM:\Software\Enterprise\Engineering`.
+
+    .EXAMPLE
+    Test-CPermission -Identity 'STARFLEET\Worf' -Permission 'Write' -ApplyTo 'Container' -Path 'C:\Enterprise\Brig'
+
+    Demonstrates how to test for inheritance/propogation flags, in addition to permissions.
+
+    .EXAMPLE
+    Test-CPermission -Identity 'STARFLEET\Data' -Permission 'GenericWrite' -Path 'cert:\LocalMachine\My\1234567890ABCDEF1234567890ABCDEF12345678'
+
+    Demonstrates how to test for permissions on a certificate's private key/key container. If the certificate doesn't
+    have a private key, returns `$true`.
+    #>
+    [CmdletBinding()]
+    param(
+        # The path on which the permissions should be checked.  Can be a file system or registry path. For certificate
+        # private keys, pass a certificate provider path, e.g. `cert:`.
+        [Parameter(Mandatory)]
+        [String] $Path,
+
+        # The user or group whose permissions to check.
+        [Parameter(Mandatory)]
+        [String] $Identity,
+
+        # The permission to test for: e.g. FullControl, Read, etc.  For file system items, use values from
+        # [System.Security.AccessControl.FileSystemRights](http://msdn.microsoft.com/en-us/library/system.security.accesscontrol.filesystemrights.aspx).
+        # For registry items, use values from
+        # [System.Security.AccessControl.RegistryRights](http://msdn.microsoft.com/en-us/library/system.security.accesscontrol.registryrights.aspx).
+        [Parameter(Mandatory)]
+        [String[]] $Permission,
+
+        # The inheritance flags to check for. Default is to ignore inheritance flags when testing the permission.
+        [InheritanceFlags] $InheritanceFlag,
+
+        # The propagation flags to check for. Default is to ignore propagation flags when testing the permission.
+        [PropagationFlags] $PropagationFlag,
+
+        # Include inherited permissions in the check.
+        [switch] $Inherited,
+
+        # Check for the exact permissions, inheritance flags, and propagation flags, i.e. make sure the identity has
+        # *only* the permissions you specify.
+        [switch] $Exact

```

Comment:
In other modules I think we use `-Strict` for this?

Reply:
I was debating whether or not to rename this. Debate closed. Renamed.